services/schemas/soldier.zschema.ts

```diff
@@ -0,0 +1,17 @@
+import { z } from 'zod';
+
+const soldier = z.object({
+  id: z.string(),
+  rank: z.string(),
+  name: z.string(),
+  limitations: z.array(z.string()),
+  duties: z.array(z.string()).optional().default([]),
+});
+
+const soldierQuery = soldier.partial();
+const soldiers = z.array(soldier);
+
+type Soldier = z.infer<typeof soldier>;
+type SoldierQuery = z.infer<typeof soldierQuery>;
+
+export { soldier, soldiers, soldierQuery, Soldier, SoldierQuery };
```

Comment:
Think of a world without typescript. If your IDE couldn't help you out, how would you expect to know what the difference between` soldier, Soldier, and soldiers` is ? Sounds far fetched but that's just how javascript is.
Clean this up by doing the following:
`soldiers` -> `SoldierSchema`. Classes and schemas should start with a capital letter.
`Soldier` -> keep it like it is.
`soldiers` -> erase it and use `Soldier []`
`soldierQuery` -> `SoldierQuerySchema`.
`SoldierQuery` -> use native mongodb native types. Ask me if you get stuck for too long.